src/components/app/data/hooks/useBFF.js

```diff
@@ -1,36 +1,37 @@
 import { useLocation, useParams } from 'react-router-dom';
 import { useQuery } from '@tanstack/react-query';
-import { resolveBFFQuery } from '../../routes/data/utils';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import { isBFFFeatureFlagEnabled } from '../utils';
+import { resolveBFFQuery } from '../queries';
 
 /**
  * Uses the route to determine which API call to make for the BFF
  * Populates the queryKey with the appropriate enterprise customer uuid once BFF call is resolved
  * @param queryOptions
+ * @param bffQueryFallback
  * @returns  {Types.UseQueryResult}} The query results for the routes BFF.
  */
-export function useBFF(queryOptions = {}) {
-  const { select, ...queryOptionsRest } = queryOptions;
+export default function useBFF(queryOptions = {}, bffQueryFallback = null) {
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+  const {
+    enabled, ...queryOptionsRest
+  } = queryOptions;
   const location = useLocation();
-  const params = useParams();
+
+  const shouldUseBFF = isBFFFeatureFlagEnabled(enterpriseCustomer.uuid);
   // Determine the BFF query to use based on the current location
+  const params = useParams();
   const matchedBFFQuery = resolveBFFQuery(location.pathname);
-  return useQuery({
+
+  // Determine which query to call, the original hook or the new BFF
+  let query = {
     ...matchedBFFQuery(params),
     ...queryOptionsRest,
-    select: (data) => {
-      if (!data) {
-        return data;
-      }
-
-      // TODO: Determine if returned data needs further transformations
-      const transformedData = structuredClone(data);
-      if (select) {
-        return select({
-          original: data,
-          transformed: transformedData,
-        });
-      }
-      return transformedData;
-    },
-  });
+    enabled: enabled && shouldUseBFF,
+  };
+  // Determine if the flag is enabled, or if we have retrieved teh fallback query
+  if ((!shouldUseBFF || matchedBFFQuery(params).queryKey.includes('fallback')) && !!bffQueryFallback) {
+    query = bffQueryFallback;
+  }
+  return useQuery(query);
```

Comment:
Just submitting a refactor I ended up prototyping locally, taking into account some of the other feedback. Can chat about it further.

```jsx
/**
 * Uses the route to determine which API call to make for the BFF
 * Populates the queryKey with the appropriate enterprise customer uuid once BFF call is resolved
 * @param queryOptions
 * @param bffQueryFallback
 * @returns  {Types.UseQueryResult}} The query results for the routes BFF.
 */
export default function useBFFQuery({
  bffQueryAdditionalParams = {},
  bffQueryOptions = {},
  fallbackQueryConfig = null,
}) {
  const { data: enterpriseCustomer } = useEnterpriseCustomer();
  const location = useLocation();
  const { enterpriseSlug } = useParams();

  // Determine the BFF query to use based on the current location
  const matchedBFFQuery = resolveBFFQuery(
    location.pathname,
    {
      enterpriseCustomerUuid: enterpriseCustomer.uuid,
      bffEnabledCustomers: getConfig().FEATURE_ENABLE_BFF_API_FOR_ENTERPRISE_CUSTOMERS,
    },
  );

  // Determine which query to call, the original hook or the new BFF
  let queryConfig = {};
  if (matchedBFFQuery) {
    queryConfig = {
      // All BFF queries should be passed the enterpriseSlug and any additional params
      ...matchedBFFQuery({ enterpriseSlug, ...bffQueryAdditionalParams }),
      ...bffQueryOptions,
    };
  } else if (fallbackQueryConfig) {
    queryConfig = fallbackQueryConfig;
  } else {
    const err = new Error('No BFF query found for the current route and no fallback query provided');
    logError(err);
    throw err;
  }

  // Call useQuery with the resolved query config
  return useQuery(queryConfig);
}
```